workspaces/ui-v2/src/optic-components/pages/diffs/AddEndpointsPage/components/BulkLearnModal.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { FC, useEffect, useState, useRef } from 'react';
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  LinearProgress,
+  Typography,
+} from '@material-ui/core';
+import { EndpointName } from '<src>/optic-components/common';
+import { SpinningOpticLogo } from '<src>/optic-components/loaders';
+
+import { useSharedDiffContext } from '<src>/optic-components/hooks/diffs/SharedDiffContext';
+import { IUndocumentedUrl } from '<src>/optic-components/hooks/diffs/SharedDiffState';
+import { makePattern } from '../utils';
+
+type BulkLearnModalProps = {
+  undocumentedEndpointsToLearn: IUndocumentedUrl[];
+  closeModal: () => void;
+};
+
+const LearningProgress: FC<{
+  pendingEndpointIds: Set<string>;
+  closeModal: BulkLearnModalProps['closeModal'];
+}> = ({ pendingEndpointIds, closeModal }) => {
+  const { stageEndpoint, context } = useSharedDiffContext();
+  const refStageEndpoint = useRef(stageEndpoint);
+
+  // @nic TODO handle when learning fails! - currently the modal just gets stuck :(
```

Comment:
There's a couple of endpoints that I've seen fail, that should be fixed but separately, a failure should not block a user from continuing here - I'm going to follow this up in a separate PR